SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -362,6 +372,76 @@ void ASpatialDebugger::OnToggleConfigUI()
 	}
 }
 
+void ASpatialDebugger::OnSelectActor()
+{
+	if (HitActors.Num() > 0)
```

Comment:
```suggestion
	if (HitActors.Num() > HoverIndex)
```

Reply:
I was validating every time I changed the index or the contents of the hit array but you are right it makes more sense to validate before attempting to access the array. 
I have encapsulated the read access with a validate which is safer.